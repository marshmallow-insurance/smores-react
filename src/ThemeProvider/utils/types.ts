import * as theme from '@mrshmllw/smores-foundations/build/web/variables.json'
import { Color } from 'theme'

type Flatten<T, Prefix extends string = ''> = {
  [K in keyof T & string]: T[K] extends Record<string, any>
    ? Flatten<T[K], `${Prefix}${K}.`>
    : `${Prefix}${K}`
}[keyof T & string]

export type NewColour = Flatten<(typeof theme)['color']>

const legacyColorMap = {
  marshmallowPink: 'interactive.primary.base', // old â†’ new
}

export type LegacyColorMap = keyof typeof legacyColorMap
