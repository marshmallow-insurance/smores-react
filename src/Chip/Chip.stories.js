import React from 'react'
import { Chip } from './Chip'

export default {
  title: 'Chip',
  component: Chip,
  argTypes: { handleClick: { action: 'clicked' } },
}

const Template = (args) => <Chip {...args} />

export const Primary = Template.bind({})

Primary.args = {
  primary: true,
  children: 'Add',
  icon: 'plus',
}

export const Secondary = Template.bind({})

Secondary.args = {
  secondary: true,
  children: 'Added',
  icon: 'tick',
}

export const Loading = Template.bind({})

Loading.args = {
  loading: true,
}

export const SecondaryDisabled = Template.bind({})

SecondaryDisabled.args = {
  secondary: true,
  disabled: true,
  loading: false,
  children: 'Add',
  icon: 'plus',
}
