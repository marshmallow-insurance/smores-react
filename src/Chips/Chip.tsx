import React, { FC, ReactNode } from 'react'
import styled, { css } from 'styled-components'

import { theme } from '../theme'
import { Icon as IconComponent } from '../Icon'

interface IButton {
  primary: boolean
  secondary: boolean
  icon: string
}

type Props = {
  children: ReactNode
  handleClick: (e: React.FormEvent<HTMLButtonElement>) => void
  primary?: boolean
  secondary?: boolean
  icon?: string
}

export const Chip: FC<Props> = ({
  children,
  handleClick,
  primary = false,
  secondary = false,
  icon = '',
}) => {
  return (
    <div>
      <Container
        primary={primary}
        secondary={secondary}
        onClick={handleClick}
        icon={icon}
      >
        {icon && (
          <IconComponent
            render={icon}
            size={20}
            color={primary ? 'white' : 'pink5'}
          />
        )}
        <ChildrenContainer>{children}</ChildrenContainer>
      </Container>
    </div>
  )
}

const Container = styled.button<IButton>(
  ({ primary, secondary, icon }) => css`
    align-items: center;
    background-color: ${theme.colors.pink5};
    border-radius: 100px;
    border-shadow: none;
    border: 2px solid;
    color: ${theme.colors.white};
    display: flex;
    font-size: 16px;
    justify-content: ${icon ? 'space-evenly' : 'center'};
    line-height: 100%;
    padding: 8px 16px 8px ${icon ? '8px' : '16px'};
    width: 98px;

    ${primary &&
    css`
      &:hover {
        background-color: ${theme.colors.pink6};
      }
    `}

    ${secondary &&
    css`
      background-color: ${theme.colors.white};
      border-color: ${theme.colors.pink5};
      color: ${theme.colors.pink5};
      &:hover {
        background-color: ${theme.colors.bg2};
      }
    `}
  `,
)

// const IconContainer = styled(IconComponent)`
//   margin-right: 10px;
// `

const ChildrenContainer = styled.div`
  flex-grow: 1;
`
