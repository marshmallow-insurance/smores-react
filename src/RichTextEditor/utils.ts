import { $createParagraphNode, LexicalNode, RootNode } from 'lexical'

export const appendNodes = (root: RootNode, nodes: LexicalNode[]) => {
  nodes
    .filter((node) => node.__type !== 'linebreak')
    .map((node) => {
      if (node.__type === 'text') {
        const paragraphNode = $createParagraphNode()
        paragraphNode.append(node)
        return paragraphNode
      }
      return node
    })
    .forEach((node) => root.append(node))
}
