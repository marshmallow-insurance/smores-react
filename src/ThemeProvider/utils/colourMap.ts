import * as designTokens from '@mrshmllw/smores-foundations/build/web/variables.json'
import { Color } from 'theme'

type Flatten<T, Prefix extends string = ''> = {
  [K in keyof T & string]: T[K] extends Record<string, any>
    ? Flatten<T[K], `${Prefix}${K}.`>
    : `${Prefix}${K}`
}[keyof T & string]

export type NewColor = Flatten<
  Pick<typeof designTokens, 'color' | 'extended1' | 'thirdParty'>
>

export type ColorTypes = Color | NewColor

// TODO: Test how well this works
// we currently access numerical colours like `theme.color.neutral[100]` or in the case of 000 colours, `theme.color.neutral['000']`
export const getThemeColor = (path: NewColor): string => {
  return path
    .split('.')
    .reduce((acc, key) => acc?.[key], designTokens as any) as string
}

// a function that returns a flattened dot notation string path to access the color value
export const formatDesignTokenColor = (): string => {
  const { color, extended1, thirdParty } = designTokens
  const colors = { color, extended1, thirdParty }

  if (colors === null) {
    return ''
  }

  const result: string[] = []
  const destructureNestedObject = (obj: any, currentPath: string) => {
    for (const key in obj) {
      if (Object.prototype.hasOwnProperty.call(obj, key)) {
        const value = obj[key]
        const newPath = currentPath ? `${currentPath}.${key}` : key
        if (typeof value === 'object' && value !== null) {
          destructureNestedObject(value, newPath)
        } else {
          result.push(newPath)
        }
      }
    }
  }
  destructureNestedObject(colors, '')
  return result.join(', ')
}

// old colour name â†’ new colour path // base token value
export const legacyColorMap: Record<Color, NewColor> = {
  fairyFloss: 'color.surface.brand.100', // palette.primary.040
  bubblegum: 'color.surface.brand.200', // palette.primary.060
  marshmallowPink: 'color.surface.brand.300', // palette.primary.100
  lollipop: 'color.surface.brand.400', // palatte.primary.120

  // Core Secondary
  chia: 'color.feedback.inactive.100', // palette.secondary.040
  sesame: 'color.text.subtle', // palette.secondary.060
  liquorice: 'color.text.base', // palette.secondary.100
  boba: 'color.text.contrast', // palette.secondary.120

  // Core Tertiary
  cream: `color.surface.base.000`, // palette.neutral.000
  coconut: 'color.surface.base.100', // palette.neutral.010
  mascarpone: 'color.surface.base.200', // palette.neutral.020
  custard: 'color.surface.base.300', // palette.neutral.040

  // Brand Secondary
  feijoa: 'color.illustration.accent1.100', // palette.brand1.060
  spearmint: 'color.illustration.accent1.200', // palette.brand1.100
  macaroon: 'color.illustration.accent2.100', // palette.brand2.060
  blueberry: 'color.illustration.accent2.200', // palette.brand2.100
  pistachio: 'color.illustration.accent3.200', // palette.brand3.100
  matcha: 'color.illustration.accent3.100', // palette.brand3.060
  caramel: 'color.illustration.accent4.200', // palette.brand4.100
  peanut: 'color.illustration.accent4.100', // palette.brand4.060
  marzipan: 'color.illustration.neutral.400', // palette.neutral.100
  oatmeal: 'color.illustration.neutral.300', // palette.neutral.060
  satsuma: 'extended1.20', // extended1.020

  // Traffic light
  watermelon: 'color.feedback.negative.100', // palette.negative.020
  strawberry: 'color.feedback.negative.200', // palette.negative.100
  apple: 'color.feedback.positive.200', // palette.positive.100
  mint: 'color.feedback.positive.100', // palette.positive.020
  lemon: 'color.feedback.notice.200', // palette.notice.100
  sherbert: 'color.feedback.notice.100', // palette.notice.020
  tangerine: 'extended1.100', // extended1.100

  // Third-party brand colours
  compareTheMarket: 'thirdParty.compareTheMarket',
  confused: 'thirdParty.confusedCom',
  onfido: 'thirdParty.onfido',
  x: 'thirdParty.twitter', // x rebrand not yet reflected in foundations
  premfina: 'thirdParty.premfina',
  checkout: 'thirdParty.checkout',
  meta: 'thirdParty.facebook', // Meta rebrand not yet reflected in foundations
  stripe: 'thirdParty.stripe',
  intercom: 'thirdParty.intercom',
  ravelin: 'thirdParty.ravelin',
  rac: 'thirdParty.rac',
  hometree: 'thirdParty.hometree',
} as const

/* usage example */
// prop change:
// change color prop type to be `color?: Color | NewColor`
// component usage:
// <Component color={'marshmallowPink'}/>
// OR
// <Component color={'interactive.primary.base'}/>
// inside component:
// const resolvedColor = color in legacyColorMap ? getThemeColor(legacyColorMap[color]) : getThemeColor(color)

// the above should resolve to #FF88C8
