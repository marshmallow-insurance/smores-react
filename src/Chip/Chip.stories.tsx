import React from 'react'
import { Chip, ChipProps } from './Chip'

export default {
  title: 'Chip',
  component: Chip,
  argTypes: { handleClick: { action: 'clicked' } },
}

const Template = (props: ChipProps) => <Chip {...props} />

export const Primary = Template.bind({})

Primary.args = {
  primary: true,
  children: 'Add',
  icon: 'plus',
  'data-testid': 'works!',
}

export const Secondary = Template.bind({})

Secondary.args = {
  secondary: true,
  children: 'Added',
  icon: 'tick',
}

export const Loading = Template.bind({})

Loading.args = {
  loading: true,
  primary: true,
}

export const SecondaryDisabled = Template.bind({})

SecondaryDisabled.args = {
  secondary: true,
  disabled: true,
  loading: false,
  children: 'Add',
  icon: 'plus',
}
