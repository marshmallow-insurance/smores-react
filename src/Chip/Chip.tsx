import React, { FC, ReactNode, ButtonHTMLAttributes, forwardRef } from 'react'
import styled, { css } from 'styled-components'

import { theme } from '../theme'
import { Box } from '../Box'
import { Icon as IconComponent } from '../Icon'
import { Loader } from '../Loader'
import { focusOutline } from '../utils/focusOutline'
import { MarginProps } from '../utils/space'

interface IButton {
  primary: boolean
  secondary: boolean
  icon: string
  $loading: boolean
  disabled: boolean
}

type Props = {
  children: ReactNode
  handleClick: (e: React.FormEvent<HTMLButtonElement>) => void
  primary?: boolean
  secondary?: boolean
  icon?: string
  disabled?: boolean
  loading?: boolean
} & MarginProps

export type ChipProps = Props & ButtonHTMLAttributes<HTMLButtonElement>

export const Chip: FC<ChipProps> = forwardRef<HTMLButtonElement, ChipProps>(
  (
    {
      children,
      handleClick,
      primary = false,
      secondary = false,
      disabled = false,
      loading = false,
      icon = '',
      ...props
    },
    ref,
  ) => (
    <Container
      as="button"
      primary={primary}
      secondary={secondary}
      disabled={disabled || loading}
      $loading={loading}
      onClick={handleClick}
      icon={icon}
      {...props}
      ref={ref}
    >
      {loading ? (
        <Loader color={primary ? 'white' : 'primary'} height="16" />
      ) : (
        <>
          {icon && (
            <IconComponent
              render={icon}
              size={20}
              color={primary ? 'white' : 'primary'}
            />
          )}
          <ChildrenContainer>{children}</ChildrenContainer>
        </>
      )}
    </Container>
  ),
)

Chip.displayName = 'Chip'

const Container = styled(Box)<IButton>(
  ({ primary, secondary, icon, $loading, disabled }) => css`
    ${focusOutline()}

    align-items: center;
    background-color: ${theme.colors.primary};
    border-radius: 100px;
    border: 2px solid ${theme.colors.primary};
    color: ${theme.colors.white};
    display: flex;
    font-size: 16px;
    justify-content: ${icon ? 'space-evenly' : 'center'};
    line-height: 100%;
    padding: 8px 16px 8px ${icon ? '8px' : '16px'};
    width: 98px;
    cursor: ${disabled || $loading ? 'not-allowed' : 'pointer'};
    opacity: ${disabled ? '0.5' : '1'};

    ${primary &&
    css`
      &:hover {
        background-color: ${(disabled || $loading) && theme.colors.primary};
      }
    `}
    ${secondary &&
    css`
      color: ${theme.colors.primary};
      background-color: ${theme.colors.white};
      &:hover {
        background-color: ${!(disabled || $loading) && theme.colors.background};
      }
    `};
  `,
)

const ChildrenContainer = styled.div`
  flex-grow: 1;
`
