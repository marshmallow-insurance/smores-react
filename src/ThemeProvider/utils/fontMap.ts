import * as designTokens from '@mrshmllw/smores-foundations/build/web/variables.json'

import { Typo } from '../../Text'
import { Theme } from '../../ThemeProvider/ThemeProvider'
import { getFromObject } from '../../utils/getFromObject'

type Font = typeof designTokens.font
type FontKey = keyof Font
export type FontValueObject = Font[keyof Font][keyof Font[keyof Font]]

const fontValueKeys = [
  'fontFamily',
  'fontSize',
  'fontWeight',
  'lineHeight',
  'letterSpacing',
  'textCase',
]

type captionTypeKeys = [keyof Font['caption']]
type bodyTypeKeys = [keyof Font['body']]
type headingTypeKeys = [keyof Font['heading']]
type heroTypeKeys = [keyof Font['hero']]
type labelTypeKeys = [keyof Font['label']]
type linkTypeKeys = [keyof Font['link']]

export type FontTypeMap = {
  caption: captionTypeKeys
  label: labelTypeKeys
  body: bodyTypeKeys
  heading: headingTypeKeys
  hero: heroTypeKeys
  link: linkTypeKeys
}[keyof Font]

// type for the path to each font type in design tokens, e.g. 'font.hero.200'
export type FontPathMap = {
  [K in FontKey]: K extends 'caption'
    ? `font.${K}.${captionTypeKeys[number]}`
    : K extends 'body'
      ? `font.${K}.${bodyTypeKeys[number]}`
      : K extends 'heading'
        ? `font.${K}.${headingTypeKeys[number]}`
        : K extends 'hero'
          ? `font.${K}.${heroTypeKeys[number]}`
          : K extends 'label'
            ? `font.${K}.${labelTypeKeys[number]}`
            : never
}[FontKey]

export type TypoTypes = FontPathMap | Typo

// a function that returns a font value object based on either the new design token path or the legacy typo mapping
export const resolveToThemeFont = (
  typo: TypoTypes,
  theme: Theme,
): FontValueObject => {
  const fontPath = getTypoPath(typo) // always returns new design token path
  return getFromObject({
    obj: theme,
    path: fontPath,
  })
}

export const getTypoPath = (typo: TypoTypes) => {
  return typo in legacyFontStyleMapping
    ? legacyFontStyleMapping[typo as keyof typeof legacyFontStyleMapping]
    : typo
}

export const formatDesignTokenFont = (): string => {
  const { font } = designTokens
  if (font === null) {
    return ''
  }

  const result: string[] = []
  const destructureNestedObject = (obj: any, currentPath: string) => {
    for (const key in obj) {
      if (Object.prototype.hasOwnProperty.call(obj, key)) {
        const value = obj[key]

        const newPath = currentPath ? `${currentPath}.${key}` : key

        // TODO: instead of trimming, we should not add paths that end with font value keys
        const trimmedPath = fontValueKeys.some((key) =>
          newPath.endsWith(`.${key}`),
        )
          ? newPath.slice(0, newPath.lastIndexOf('.'))
          : newPath

        if (typeof value === 'object' && value !== null) {
          destructureNestedObject(value, trimmedPath)
        } else {
          result.push(trimmedPath)
        }
      }
    }
  }
  destructureNestedObject(font, 'font')
  return result.join(', ')
}

export const newFontStyles: Font = designTokens.font

export const legacyFontStyleMapping: Record<Typo, FontPathMap> = {
  'hero-alternate': 'font.hero.300',
  hero: 'font.hero.200',
  'heading-alternate': 'font.hero.100',
  'heading-large': 'font.heading.500',
  'heading-medium': 'font.heading.300',
  'heading-small': 'font.heading.200',
  'headline-regular': 'font.heading.100',
  'headline-small': 'font.heading.100', // query this
  'body-standfirst': 'font.body.300',
  'body-regular': 'font.body.200',
  'body-small': 'font.body.100',
  caption: 'font.caption.100',
  label: 'font.label.100',
}
