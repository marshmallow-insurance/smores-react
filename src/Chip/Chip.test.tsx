import React from 'react'
import { expect, it } from 'vitest'
import { render } from '@testing-library/react'
import { Chip } from './Chip'
import { noop } from '../utils/noop'

describe('Chip', () => {
  it('renders correctly with primary styling', () => {
    const { container } = render(
      <Chip handleClick={noop} primary>
        Primary
      </Chip>,
    )
    expect(container).toMatchSnapshot()
  })
  it('renders correctly with secondary styling', () => {
    const { container } = render(
      <Chip handleClick={noop} secondary>
        Secondary
      </Chip>,
    )
    expect(container).toMatchSnapshot()
  })
  it('renders correctly when disabled', () => {
    const { container } = render(
      <Chip handleClick={noop} disabled>
        Disabled
      </Chip>,
    )
    expect(container).toMatchSnapshot()
  })
  it('renders correctly when loading', () => {
    const { container } = render(
      <Chip handleClick={noop} loading>
        Loading
      </Chip>,
    )
    expect(container).toMatchSnapshot()
  })
})
