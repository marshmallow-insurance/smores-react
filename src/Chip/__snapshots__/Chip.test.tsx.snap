// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Chip > renders correctly when disabled 1`] = `
.c0 {
  outline: 0;
  align-items: center;
  background-color: #f1e9dc;
  border-radius: 100px;
  border: 2px solid #f1e9dc;
  color: #292924;
  display: flex;
  font-size: 16px;
  justify-content: center;
  line-height: 100%;
  padding: 8px 16px 8px 16px;
  min-width: 98px;
  min-height: 40px;
  cursor: not-allowed;
  opacity: 0.5;
}

.c0:focus-visible {
  outline: 2px solid #292924;
  outline-offset: 2px;
}

.c1 {
  flex-grow: 1;
}

<div>
  <button
    aria-label="chip-button"
    class="c0"
    disabled=""
  >
    <div
      class="c1"
    >
      Disabled
    </div>
  </button>
</div>
`;

exports[`Chip > renders correctly when loading 1`] = `
.c1 path {
  fill: #ffffff;
  stroke: none;
  fill-rule: evenodd;
}

.c2 {
  animation: Marshmallow1_t 1.6s linear infinite both,Marshmallow1_d 1.6s linear infinite both;
}

.c3 {
  animation: Marshmallow2_t 1.6s linear infinite both,Marshmallow2_d 1.6s linear infinite both;
}

.c4 {
  animation: Marshmallow3_t 1.6s linear infinite both,Marshmallow3_d 1.6s linear infinite both;
}

.c0 {
  outline: 0;
  align-items: center;
  background-color: #f1e9dc;
  border-radius: 100px;
  border: 2px solid #f1e9dc;
  color: #292924;
  display: flex;
  font-size: 16px;
  justify-content: center;
  line-height: 100%;
  padding: 8px 16px 8px 16px;
  min-width: 98px;
  min-height: 40px;
  cursor: not-allowed;
  opacity: 0.5;
}

.c0:focus-visible {
  outline: 2px solid #292924;
  outline-offset: 2px;
}

<div>
  <button
    aria-label="chip-button"
    class="c0"
    disabled=""
  >
    <svg
      class="c1"
      height="16"
      viewBox="0 0 60 32"
    >
      <g
        transform="translate(30,18.5455) translate(-30,-9)"
      >
        <path
          class="c2"
          d="M73.999,81.9767L73.6592,70.6087C73.6352,69.8071,73.1892,69.0806,72.4916,68.707C71.1747,68.0014,69.0105,67.648,66.0001,67.6467C62.9894,67.648,60.8254,68.0014,59.5084,68.707C58.8109,69.0806,58.3648,69.8071,58.3408,70.6087L58.001,81.9767C57.983,82.5754,58.2138,83.1538,58.6366,83.5705C60.0408,84.9546,62.4906,85.6467,65.986,85.6467C65.9907,85.6467,65.9953,85.6467,66.0001,85.6467C66.0047,85.6467,66.0094,85.6467,66.014,85.6467C69.5095,85.6467,71.9593,84.9546,73.3634,83.5705C73.7863,83.1538,74.0169,82.5754,73.999,81.9767"
          id="Marshmallow1"
          transform="translate(-58,-67.6467)"
        />
        <path
          class="c3"
          d="M73.999,81.9767L73.6592,70.6087C73.6352,69.8071,73.1892,69.0806,72.4916,68.707C71.1747,68.0014,69.0105,67.648,66.0001,67.6467C62.9894,67.648,60.8254,68.0014,59.5084,68.707C58.8109,69.0806,58.3648,69.8071,58.3408,70.6087L58.001,81.9767C57.983,82.5754,58.2138,83.1538,58.6366,83.5705C60.0408,84.9546,62.4906,85.6467,65.986,85.6467C65.9907,85.6467,65.9953,85.6467,66.0001,85.6467C66.0047,85.6467,66.0094,85.6467,66.014,85.6467C69.5095,85.6467,71.9593,84.9546,73.3634,83.5705C73.7863,83.1538,74.0169,82.5754,73.999,81.9767"
          id="Marshmallow2"
          transform="translate(-36,-67.6467)"
        />
        <path
          class="c4"
          d="M73.999,81.9767L73.6592,70.6087C73.6352,69.8071,73.1892,69.0806,72.4916,68.707C71.1747,68.0014,69.0105,67.648,66.0001,67.6467C62.9894,67.648,60.8254,68.0014,59.5084,68.707C58.8109,69.0806,58.3648,69.8071,58.3408,70.6087L58.001,81.9767C57.983,82.5754,58.2138,83.1538,58.6366,83.5705C60.0408,84.9546,62.4906,85.6467,65.986,85.6467C65.9907,85.6467,65.9953,85.6467,66.0001,85.6467C66.0047,85.6467,66.0094,85.6467,66.014,85.6467C69.5095,85.6467,71.9593,84.9546,73.3634,83.5705C73.7863,83.1538,74.0169,82.5754,73.999,81.9767"
          id="Marshmallow3"
          transform="translate(-14,-67.6467)"
        />
      </g>
    </svg>
  </button>
</div>
`;

exports[`Chip > renders correctly with primary styling 1`] = `
.c0 {
  outline: 0;
  align-items: center;
  background-color: #f1e9dc;
  border-radius: 100px;
  border: 2px solid #f1e9dc;
  color: #292924;
  display: flex;
  font-size: 16px;
  justify-content: center;
  line-height: 100%;
  padding: 8px 16px 8px 16px;
  min-width: 98px;
  min-height: 40px;
  cursor: pointer;
  opacity: 1;
}

.c0:focus-visible {
  outline: 2px solid #292924;
  outline-offset: 2px;
}

.c0:hover {
  border: 2px solid #f8f2ea;
  background-color: #f8f2ea;
}

.c1 {
  flex-grow: 1;
}

<div>
  <button
    aria-label="chip-button"
    class="c0"
  >
    <div
      class="c1"
    >
      Primary
    </div>
  </button>
</div>
`;

exports[`Chip > renders correctly with secondary styling 1`] = `
.c0 {
  outline: 0;
  align-items: center;
  background-color: #f1e9dc;
  border-radius: 100px;
  border: 2px solid #f1e9dc;
  color: #292924;
  display: flex;
  font-size: 16px;
  justify-content: center;
  line-height: 100%;
  padding: 8px 16px 8px 16px;
  min-width: 98px;
  min-height: 40px;
  cursor: pointer;
  opacity: 1;
  color: #ffffff;
  background-color: #292924;
  border: 2px solid #292924;
}

.c0:focus-visible {
  outline: 2px solid #292924;
  outline-offset: 2px;
}

.c0:hover {
  border: 2px solid #636768;
  background-color: #636768;
}

.c1 {
  flex-grow: 1;
}

<div>
  <button
    aria-label="chip-button"
    class="c0"
  >
    <div
      class="c1"
    >
      Secondary
    </div>
  </button>
</div>
`;
